python_modules/dagster/dagster/_core/definitions/declarative_automation/automation_condition.py

```diff
@@ -512,13 +515,27 @@ def any_downstream_conditions() -> "AnyDownstreamConditionsCondition":
         return AnyDownstreamConditionsCondition()
 
 
-class AutomationResult:
+class BuiltinAutomationCondition(AutomationCondition[T_EntityKey], DagsterModel):
```

Comment:
hmmmm why are we using `DagsterModel` here? Assume we are eliminating that.

Reply:
This was basically a workaround for the generic record issue + an additional record subclassing issue (both tracked). The generic issue was fixed, going to take a crack at the subclassing one, after which these can get converted to records

Reply:
this is spitting out a bunch of warnings as is 
https://gist.github.com/alangenfeld/e64b772d77d8aefe1a68cf8d297081e8
so if `@record` doesn't work out well need to tweak something to address that